fetch/metrics/hash_to_peers_cache.go

```diff
@@ -0,0 +1,34 @@
+package metrics
+
+import (
+	"github.com/spacemeshos/go-spacemesh/metrics"
+)
+
+const (
+	// subsystem shared by all metrics exposed by this package.
+	subsystem = "fetch"
+)
+
+var (
+	totalHits   = metrics.NewCounter("total_hits", subsystem, "Total cache hits", nil)
+	totalMisses = metrics.NewCounter("total_misses", subsystem, "Total cache misses", nil)
+)
+
+// HashToPeersCacheCollector implement metrics.Reporter
+// that keeps track of the number of messages sent and received per protocol.
+type HashToPeersCacheCollector struct{}
+
+// NewHashToPeersCacheCollector creates a new HashToPeersCacheCollector.
+func NewHashToPeersCacheCollector() *HashToPeersCacheCollector {
```

Comment:
even if you need an object for some reason, what the point of empty initializer? why would someone even use it instead of zero value?

Reply:
there was a point but not after simplifying it :) removed